lib/cli/run.js

```diff
@@ -0,0 +1,403 @@
+        group: GROUPS.FILES,
+        requiresArg: true,
+        type: 'array'
+      },
+      retries: {
+        defaultDescription: '0',
+        description: 'Retry failed tests this many times',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      slow: {
+        alias: 's',
+        default: 75,
+        description: 'Specify "slow" test threshold (in milliseconds)',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      sort: {
+        alias: 'S',
+        description: 'Sort test files',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      },
+      timeout: {
+        alias: 't',
+        default: 2000,
+        description: 'Specify test timeout threshold (in milliseconds)',
+        group: GROUPS.RULES
+      },
+      ui: {
+        alias: 'u',
+        default: 'bdd',
+        description: 'Specify user interface',
+        group: GROUPS.RULES,
+        requiresArg: true,
+        type: 'string'
+      },
+      watch: {
+        alias: 'w',
+        description: 'Watch files in the current working directory for changes',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      }
+    })
+    .positional('spec', {
+      default: ['test/'],
+      defaultDescription: 'test',
+      description: 'One or more files, directories, or globs to test',
+      type: 'array'
+    })
+    .config('config', 'Load config file', filepath => {
+      // yargs normalizes this into an absolute path, which is why
+      // this is weird.
+      if (filepath === path.resolve(CONFIG_PATH_PLACEHOLDER)) {
+        debug(`looking for config from ${cwd}`);
+        return findConfig(cwd);
+      }
+      debug(`loading config at ${filepath}`);
+      return loadConfig(filepath);
+    })
+    .pkgConf('mocha')
+    .check(argv => {
+      if (argv.invert && !('fgrep' in argv || 'grep' in argv)) {
+        throw new Error(
+          '"--invert" requires one of "--fgrep <str>" or "--grep <regexp>"'
+        );
+      }
+
+      validatePlugin(argv, 'reporter', Mocha.reporters);
+      validatePlugin(argv, 'ui', Mocha.interfaces);
+
+      return true;
+    });
+
+exports.handler = argv => {
+  debug('post-yargs config', argv);
+
+  if (argv.interfaces) {
+    debug('interfaces:');
+    runShowInterfaces();
+    process.exit();
+  }
+  if (argv.reporters) {
+    debug('reporters:');
+    runShowReporters();
+    process.exit();
+  }
+
+  // load requires first, because it can impact subsequent operations.
+  handleRequires(argv);
+
+  const mocha = new Mocha();
+
+  mocha
+    .bail(argv.bail)
+    .fgrep(argv.fgrep)
+    .globals(argv.globals)
+    .grep(argv.grep)
+    .reporter(argv.reporter, argv.reporterOptions)
+    .retries(argv.retries)
+    .slow(argv.slow)
+    .timeout(argv.timeout)
+    .ui(argv.ui)
+    .useColors(argv.colors)
+    .useInlineDiffs(argv.inlineDiffs);
+
+  // pesky inverted options
+  mocha.hideDiff(!argv.diff).enableTimeouts(!argv.noTimeouts);
```

Comment:
We should add `showDiff` method and `timeouts (default: true)` option to allow us to use positive logic.

Reply:
agree, but tangential